### PR TITLE
Add state of Rust survey 2016

### PR DESCRIPTION
**Pull request by jonathandturner**
This is the blog post for the "State of Rust Survey 2016"

r? @rust-lang/core 
cc @skade @carols10cents @bstrie @Manishearth @erickt @brson

### CODE REVIEW

**Comment by steveklabnik** [comment]
This is exceedingly minor, but usually it's one space after the period, not two, and 80 cols. I'm not sure if it's worth the time to fix....

**Comment by jonathandturner** [comment]
@steveklabnik - that's how you know I'm old school ;)

File: _posts/2016-06-30-State-of-Rust-Survey-2016.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+We see strong numbers in project size as developers put more time into Rust at work.  Over half of the Rust users using Rust full-time at work have codebases that are tens or hundreds of thousands of lines of code.
+
+Equally encouraging is the growth we expect to see in Rust in the workplace, as we see in the next chart.
+
+<center>
+![Using Rust at work in future][rust_at_work_future]
+</center>
+
+Of those not currently using Rust at work, more than 40% plan to being able to use Rust at work.  This will help carry Rust to more places and in more areas.  Speaking of carrying to more areas, we saw a wide variety of job domains represented in the survey:
```
Comment (steveklabnik) [inline _posts/2016-06-30-State-of-Rust-Survey-2016.md]: "plan on being able to use"

**Comment by erickt** [comment]
I love it!

File: _posts/2016-06-30-State-of-Rust-Survey-2016.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+Investing in IDE support not only helps new users but also helps enable teams moving to Rust and the growth of larger codebases, as we see in some of the quotes about Rust's challenges:
+
+> "I won't use the language until there's IDE support for it, and I know other devs that feel the same way. As productive as your language's syntax is, I'm more productive in a worse language with an editor that has code-completion."
+
+> "Users/projects considering switching languages often are not willing to sacrifice tooling quality even for a better language."
+
+> "Proper IDE support (hard to get it accepted at work for that reason)"
+
+
```
Comment (steveklabnik) [inline _posts/2016-06-30-State-of-Rust-Survey-2016.md]: extremely minor: there's an extra newline here

**Comment by steveklabnik** [comment]
:+1:

**Comment by brson** [comment]
:+1:

**Comment by alexcrichton** [comment]
My minimum threshold for surveys is 3087 respondents, so we may want to rethink all our analysis here.

In other words, looks great to me!

**Pull request merged**